frame/staking/src/lib.rs

```diff
@@ -1364,14 +1601,175 @@ impl<T: Trait> Module<T> {
 		// available yet.
 		CurrentEraStartSessionIndex::put(0);
 		BondedEras::mutate(|bonded| bonded.push((0, 0)));
-		Self::select_validators().1
+		Self::select_and_update_validators()
+	}
+
+	/// Checks a given solution and if correct and improved, writes it on chain as the queued result
+	/// of the next round. This may be called by both a signed and an unsigned transaction.
+	fn check_and_replace_solution(
```

Comment:
This function is **EXTREMELY** critical to be checked for both 

1- no way to sneak in a wrong solution with any bias or misbehaviour 
2- it should be as efficient as possible.

Reply:
Can you please split this function into multiple smaller functions? This will increase the readability extremely.

Reply:
sure, but maybe in a followup or at the very end. So far this function is the center of all the work. Splitting it would make things more complicated. There is a _extensive_ document in the associated dispatchable `submit_election_result` that explains it pretty much line by line.